From d9fff37cd624f94cde45a557cf23db851471caac Mon Sep 17 00:00:00 2001
From: Abhay Krishna Arunachalam <arnchlm@amazon.com>
Date: Thu, 2 Feb 2023 01:39:15 -0800
Subject: [PATCH 3/10] Snow AMI support

- Add instance metadata options to Packer config
- Rename Snow node image to reflect appropriate CAPI provider

Signed-off-by: Abhay Krishna Arunachalam <arnchlm@amazon.com>
---
 images/capi/packer/ami/packer.json | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/images/capi/packer/ami/packer.json b/images/capi/packer/ami/packer.json
index 27a4ebfd6..cab705721 100644
--- a/images/capi/packer/ami/packer.json
+++ b/images/capi/packer/ami/packer.json
@@ -4,7 +4,7 @@
       "access_key": "{{user `aws_access_key`}}",
       "ami_description": "{{user `ami_description`}}",
       "ami_groups": "{{user `ami_groups`}}",
-      "ami_name": "capa-ami-{{user `build_name`}}-{{user `kubernetes_semver` | clean_resource_name}}-{{user `build_timestamp`}}",
+      "ami_name": "capas-ami-{{user `build_name`}}-{{user `kubernetes_semver` | clean_resource_name}}-{{user `build_timestamp`}}",
       "ami_product_codes": "",
       "ami_regions": "{{user `ami_regions`}}",
       "ami_users": "{{user `ami_users`}}",
@@ -22,6 +22,11 @@
           "volume_type": "{{ user `volume_type` }}"
         }
       ],
+      "metadata_options": {
+        "http_endpoint": "{{ user `http_endpoint` }}",
+        "http_tokens": "{{ user `http_tokens` }}",
+        "http_put_response_hop_limit": "{{ user `http_put_response_hop_limit` }}"
+      },
       "name": "{{user `build_name`}}",
       "profile": "{{ user `aws_profile`}}",
       "region": "{{ user `aws_region` }}",
@@ -168,6 +173,9 @@
     "crictl_version": null,
     "encrypted": "false",
     "existing_ansible_ssh_args": "{{env `ANSIBLE_SSH_ARGS`}}",
+    "http_endpoint": "enabled",
+    "http_tokens": "required",
+    "http_put_response_hop_limit": "2",
     "iam_instance_profile": "",
     "ib_version": "{{env `IB_VERSION`}}",
     "iops": "3000",
-- 
2.34.1

